packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -6,6 +8,44 @@ const ADVANCED_GROUP = 'advanced';
 const DISPLAY_GROUP = 'stats';
 const ZERO_CONFIG_GROUP = 'zero-config';
 
+// Extract all the flages being exported from core. A list of cli flags generated by core
+// can be found here https://github.com/webpack/webpack/blob/master/test/__snapshots__/Cli.test.js.snap
+let flagsFromCore =
+    typeof cli !== 'undefined'
+        ? Object.entries(cli.getArguments()).map(([flag, meta]) => {
+              if (meta.simpleType === 'string') {
+                  meta.type = String;
+                  meta.usage = `--${flag} <value>`;
+              } else if (meta.simpleType === 'number') {
+                  meta.type = Number;
+                  meta.usage = `--${flag} <value>`;
+              } else {
+                  meta.type = Boolean;
+                  meta.negative = true;
+                  meta.usage = `--${flag}`;
+              }
+              return {
+                  ...meta,
+                  name: flag,
+                  group: 'core',
+              };
+          })
+        : [];
+
+// duplicate flags
+const duplicateFlags = ['entry', 'mode', 'stats', 'watch', 'target'];
+// broken flags
+const brokenFlags = ['module-no-parse', 'module-no-parse-reset', 'optimization-no-emit-on-errors', 'watch-options-poll'];
+// regExp Flags
+const regExpFlags = ['module-expr-context-reg-exp', 'module-unknown-context-reg-exp', 'module-wrapped-context-reg-exp'];
+// inFeasibleFlags https://github.com/webpack/webpack-cli/pull/1630#issuecomment-646992322
+const inFeasibleFlags = flagsFromCore.filter(({ name }) => {
+    return name.includes('module-rules-') || name.includes('alias-') || name.startsWith('output-library');
+});
```

Comment:
/cc @snitin315 Why we can't handle `output-library` and `alias-`?

Reply:
`output-library` related flags some I think are broken (like `--output-library-amd`) some depends on other flags for example we can't use `--output-library-name` alone if did webpack will throw error it should be done like  
`webpack --output-library-name testName --outpyt-library-type var` .

Reply:
I think such nested webpack config not feasable via CLI hence I have refered them as inFeasableFlags. We can discuss more about it. https://github.com/webpack/webpack-cli/pull/1630#issuecomment-646992322

Reply:
/cc @sokra What do you think about it?

Reply:
I think the cli should not apply additional filtering. If some arguments exposed by webpack doesn't make sense that should be filtered on webpack side. Webpack allows to filter arguments in the Schema with `cli.exclude`. This is already done for some options, but we can add more.

Reply:
@sokra Agree, but I think some options from array still can be handled

@snitin315 Can you provide full list of filtered options from webpack? I think we can handle aliases too, but in some difference way

Reply:
Below is the full list -
```js
module-rules-compiler
module-rules-enforce
module-rules-exclude
module-rules-include
module-rules-issuer
module-rules-loader
module-rules-mimetype
module-rules-options
module-rules-real-resource
module-rules-resource
module-rules-resource-query
module-rules-side-effects
module-rules-test
module-rules-type
module-rules-use-ident
module-rules-use-loader
module-rules-use-options
module-rules-use
module-rules-reset
output-library
output-library-reset
output-library-amd
output-library-commonjs
output-library-root
output-library-root-reset
output-library-auxiliary-comment
output-library-auxiliary-comment-amd
output-library-auxiliary-comment-commonjs
output-library-auxiliary-comment-commonjs2
output-library-auxiliary-comment-root
output-library-export
output-library-export-reset
output-library-name
output-library-name-reset
output-library-name-amd
output-library-name-commonjs
output-library-name-root
output-library-name-root-reset
output-library-type
output-library-umd-named-define
resolve-alias-alias
resolve-alias-name
resolve-alias-only-module
resolve-alias-reset
resolve-alias-fields
resolve-alias-fields-reset
resolve-loader-alias-alias
resolve-loader-alias-name
resolve-loader-alias-only-module
resolve-loader-alias-reset
resolve-loader-alias-fields
resolve-loader-alias-fields-reset
```

Reply:
I think `resolve-*` can be supported with some limitations, we need implement `separator`, for example:
```

webpack resolve-alias=alias|name
```

`resolve-alias-fields` also can be easy implemented, it is just `webpack --resolve-alias-fields=foo --resolve-alias-fields=bar`, not sure what mean `resolve-alias-fields-reset`? Can you clarify?

Same for `output-library`

Reply:
`resolve-alias-fields-reset` will reset `resolve.aliasFields` to empty array. 
```
resolve: {
aliasFields: []
}
```

Reply:
Yes, but here other two cases:
```
webpack --resolve-alias=alias1|name1 --resolve-alias=alias2|name2
```

Adding `alias1` and `alias2` to aliases, you can already have aliases in your configuration

```
webpack --resolve-alias=alias1|name1 --resolve-alias=alias2|name2 --resolve-alias-fields-reset
```

Adding `alias1` and `alias2` to aliases, but reset them before, so aliases from the configuration are not respected

Reply:
> webpack --resolve-alias=alias1|name1 --resolve-alias=alias2|name2

**It's result will be -
`resolve : { alias: [ { alias: ['alias1',  'alias2'], name: ['alias1', 'alias2'] } ] }`** 

> webpack --resolve-alias=alias1|name1 --resolve-alias=alias2|name2 --resolve-alias-fields-reset

**It's result will be -
`resolve : { alias: [ { alias: ['alias1',  'alias2'], name: ['alias1', 'alias2'] } ] , allFields: [] }`**

Reply:
We need improve our arg parser, we have case where you have complex options and you can use it multiple times, any ideas?

Reply:
maybe something like `webpack --mupltiple-values=value1,value2,value3` ?

Reply:
/cc @webpack/cli-team need suggestions.

Reply:
@evilebottnawi are these options popular?

Reply:
>We need improve our arg parser, we have case where you have complex options and you can use it multiple times, any ideas?

I found commander.js will support `variadic args` https://github.com/tj/commander.js/pull/1250  in next release, This is maybe helpful.